File: Laba_1_sharp/Laba_1/Calculator.cs
using System;

namespace Laba_1
{
    public class Calculator
    {
        public double Sum { get; private set; }
        public int Count { get; private set; }

        private SingleBreaker _breaker;
        private MultipleBreaker _multipleBreaker;
        private bool _isSingleBreaker;
        private int _step;
        private int _calculateTime;
        private int _id;

        public Calculator(SingleBreaker breaker, CalculateInfo info)
        {
            _breaker = breaker;
            _step = info.Step;
            _calculateTime = info.CalculationTime_ms;
            _id = info.Id;
            _isSingleBreaker = true;
        }

        public Calculator(MultipleBreaker breaker, CalculateInfo info)
        {
            _multipleBreaker = breaker;
            _step = info.Step;
            _calculateTime = info.CalculationTime_ms;
            _id = info.Id;
            _isSingleBreaker = false;
        }

        public void Calculate()
        {
            Sum = 0;
            Count = 0;
            bool condition;

            do
            {
                Sum += _step;
                Count++;

                condition = _isSingleBreaker ? !_breaker.IsStoped : !_multipleBreaker.GetStopper(_id);
            } while (condition);

            Console.WriteLine($"Thread Id: {_id}; Sum: {Sum}; Step: {_step}; Count: {Count}; Calculation time: {_calculateTime} ms");
        }
    }
}

File: Laba_1_java/src/Calculator.java
public class Calculator extends Thread {
    private Double _sum;
    private Integer _count;
    private Breaker _breaker;
    private Integer _step;
    private Integer _calculationTime;
    private Integer _id;

    public Calculator(Breaker breaker, CalculationInfo info) {
        _breaker = breaker;
        _step = info.GetStep();
        _calculationTime = info.GetCalculationTime_ms();
        _id = info.GetId();
    }

    @Override
    public void run(){
        Calculate();
    }

    public void Calculate(){
        _sum = 0d;
        _count = 0;

        do{
            _sum += _step;
            _count++;
        }while(!_breaker.GetStopper(_id));

        System.out.println("Thread Id: " + _id + "; Sum: " + _sum + "; Step: " + _step + "; Count: " + _count + "; Calculation time: " + _calculationTime + " ms");
    }

    public Double GetSum(){
        return _sum;
    }

    public Integer GetCount(){
        return _count;
    }
}